vlasovsolver/cpu_moments.cpp

```diff
@@ -176,6 +176,9 @@ void calculateMoments_R(
           if (cell->sysBoundaryFlag == sysboundarytype::DO_NOT_COMPUTE) {
              continue;
           }
+          if (cell->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY && cell->sysBoundaryLayer != 1 && P::tstep != 0) { // these should have been handled by the boundary code
```

Comment:
You probably don't want the `&& P::tstep != 0` in here? (or in similar places below)